classes/install/Upgrade.inc.php

```diff
@@ -1930,6 +1930,277 @@ function removeCancelledReviewAssignments() {
+								<h3>'.$customItem['title'].'</h3>
+								<p>'.nl2br($customItem['content']).'</p>
+								</div>';
+						}
+					}
+				}
+				// finaly, concatenate the other settings
+				foreach ($otherSettings as $divId => $otherSetting) {
+					if ($divId == 'sponsors') {
+						if (!empty($otherSetting[$locale]) || !empty($sponsors)) {
+							$aboutJournal[$locale] .= '
+								<div id="'.$divId.'">
+								<h3>Sponsors</h3>'; // hard coded because the locale key does no exist any more
+							if (!empty($otherSetting[$locale])) {
+								$aboutJournal[$locale] .= '
+								<p>'.nl2br($otherSetting[$locale]).'</p>';
+							}
+							if (!empty($sponsors)) {
+								$aboutJournal[$locale] .= '<ul>';
+								foreach ($sponsors as $sponsor) {
+									$aboutJournal[$locale] .= '<li>';
+									if (!empty($sponsor['url'])) {
+										$aboutJournal[$locale] .= '
+											<a href="'.htmlspecialchars($sponsor['url']).'">'.htmlspecialchars($sponsor['institution']).'</a>';
+									} else {
+										$aboutJournal[$locale] .= htmlspecialchars($sponsor['institution']);
+									}
+									$aboutJournal[$locale] .= '</li>';
+								}
+								$aboutJournal[$locale] .= '</ul>';
+							}
+							$aboutJournal[$locale] .= '</div>';
+						}
+					} elseif ($divId == 'contributors') {
+						if (!empty($otherSetting[$locale]) || !empty($contributors)) {
+							$aboutJournal[$locale] .= '
+								<div id="'.$divId.'">
+								<h3>'.__($otherSetting['localKey'], array(), $locale).'</h3>';
+							if (!empty($otherSetting[$locale])) {
+								$aboutJournal[$locale] .= '
+									<p>'.nl2br($otherSetting[$locale]).'</p>';
+							}
+							if (!empty($contributors)) {
+								$aboutJournal[$locale] .= '<ul>';
+								foreach ($contributors as $contributor) {
+									$aboutJournal[$locale] .= '<li>';
+									if (!empty($contributor['url'])) {
+										$aboutJournal[$locale] .= '
+											<a href="'.htmlspecialchars($contributor['url']).'">'.htmlspecialchars($contributor['name']).'</a>';
+									} else {
+										$aboutJournal[$locale] .= htmlspecialchars($contributor['name']);
+									}
+									$aboutJournal[$locale] .= '</li>';
+								}
+								$aboutJournal[$locale] .= '</ul>';
+							}
+							$aboutJournal[$locale] .= '</div>';
+						}
+					} else {
+						if (!empty($otherSetting[$locale])) {
+							$aboutJournal[$locale] .= '
+								<div id="'.$divId.'">
+								<h3>'.__($otherSetting['localKey'], array(), $locale).'</h3>
+								<p>'.nl2br($otherSetting[$locale]).'</p>
+								</div>';
+						}
+					}
+				}
+			}
+			$journalSettingsDao->updateSetting($journal->getId(), 'about', $aboutJournal, 'string', true);
+			unset($journal);
+		}
+		return true;
+	}
+
+	/**
+	 * For 2.4.x - 3.1.0 upgrade: concatenate editorialTeam and displayMembership page to new journal setting 'masthead'.
+	 * @return boolean
+	 */
+	function concatenateIntoMasthead() {
+		$roleIdEditor = 0x00000100;
```

Comment:
They are defined in a function above, so I thought it is more secure like this, depending on where and when they are needed, which function is when executed... to have them defined in the appropriate function... OK?